Model/BalancepayMethod.php

```diff
@@ -229,13 +238,16 @@ public function __construct(
     /**
      * Check whether payment method can be used
      *
-     * @param \Magento\Quote\Api\Data\CartInterface|null $quote
+     * @param CartInterface|null $quote
      * @return bool
      * @deprecated 100.2.0
      */
-    public function isAvailable(\Magento\Quote\Api\Data\CartInterface $quote = null)
+    public function isAvailable(CartInterface $quote = null)
     {
-        if ($quote && $quote->isMultipleShippingAddresses()) {
+        $currentCustomerGroup = $this->customerSession->getCustomer()->getGroupId();
+        $allowedCustomerGroups = $this->balancepayConfig->getAllowedCustomerGroups();
+        $isCustomerGroupAllowed = in_array($currentCustomerGroup, $allowedCustomerGroups);
+        if (($quote && $quote->isMultipleShippingAddresses()) || !$isCustomerGroupAllowed) {
```

Comment:
`$quote && $quote->isMultipleShippingAddresses()` – what do you check here?

Reply:
consider:
`$isMultipleShippingAddresses = $qoute && $qoute->isMultipleShippingAdresses();`